org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/fix/OverriddenAssignmentCleanUp.java

```diff
@@ -159,42 +159,39 @@ public boolean visit(final VariableDeclarationStatement node) {
 				return true;
 			}
 
-			private boolean canMoveNls(VariableDeclarationStatement node, Assignment overridingAssignment) {
-				try {
-					return isOnOwnLine(node) && isOnOwnLine(overridingAssignment);
-				} catch (JavaModelException e) {
-					// can't guarantee correctness, so bail
-					return false;
+			/**
+			 * Check that statement containing ASTNode does not share lines with other statements
+			 *
+			 * @param node ASTNode
+			 * @return true if node's statement does not share lines with other statements otherwise false
+			 */
+			private boolean doesNotShareLines(ASTNode node) {
+				Statement stmt= null;
+				if (node instanceof Statement) {
+					stmt= (Statement)node;
+				} else {
+					stmt= (Statement)ASTNodes.getFirstAncestorOrNull(node, Statement.class);
 				}
-			}
-
-			private boolean isOnOwnLine(ASTNode node) throws JavaModelException {
-				// make sure there is only whitespace before and after the node (or a line comment)
-				int startLine= unit.getLineNumber(node.getStartPosition());
-				int endLine= unit.getLineNumber(node.getStartPosition() + node.getLength());
-				if (startLine != endLine) {
+				if (stmt == null) {
 					return false;
 				}
-				int lineStart= unit.getPosition(startLine, 0);
-				int lineEnd= unit.getPosition(startLine + 1, 0);
-				if (lineEnd < 0) {
-					lineEnd= unit.getLength();
-				}
-				String textBefore= cu.getBuffer().getText(lineStart, node.getStartPosition()-lineStart);
-				if (!textBefore.isBlank()) {
-					return false;
+				int stmtStartLine= unit.getLineNumber(stmt.getStartPosition());
+				int stmtEndLine= unit.getLineNumber(stmt.getStartPosition() + stmt.getLength());
+				Statement prevStmt= ASTNodes.getPreviousSibling(stmt);
+				if (prevStmt == null) {
+					ASTNode parent= stmt.getParent();
```

Comment:
I'm not really which condition you're trying to catch here: we're in the "this is the first statement in the block" case, right? So if we go up the parent chain, we'll end up with a method declaration, for example, no? Or are we looking at other types of blocks like this?
```
if (foo == false) { String foo= bar; //NON-NLS-1
``